### ISSUE TITLE
lvol doesn't recognise physical volume parameter

### ISSUE THREAD

**Issue by hryamzik** [ansible/ansible-modules-extras#1870]
##### Issue Type:
- Bug Report
- Feature Idea

Not sure =)
##### Plugin Name:

lvol
##### Ansible Version:

2.0.1.0
##### Summary:

lvol doesn't recognise physical volume parameter.

Here's what `lvcreate` documentation says:

> lvcreate  [--come options with keys here]  VolumeGroup{Name|Path}[/ThinPoolLogicalVolumeName]   [PhysicalVol‐
>        umePath[:PE[-PE]]...]
##### Steps To Reproduce:

```
- lvol: vg=U1458254646I0 lv=ssd_cache size=233062633s opts=/dev/sdb1
```
##### Expected Results:

```
TASK [dm-cache : create cache lv] ************************************************
changed: [test]
```
##### Actual Results:

```
TASK [dm-cache : create cache lv] ************************************************
fatal: [test]: FAILED! => {"changed": false, "err": "  Volume group \"sdb1\" not found\n", "failed": true, "msg": "Creating logical volume 'ssd_cache' failed", "rc": 5}
```
##### Workaround:

``` yml
- name: create cache lv
  lvol: vg={{ hdd_vg.stdout }} lv=ssd_cache size={{ ssd_cache_sectors }}s opts="{{ hdd_vg.stdout }} /dev/{{ ssd_drive }}"
  ignore_errors: true
  register: cache_vol

- name: create cache lv
  lvol: vg='' lv=ssd_cache size={{ ssd_cache_sectors }}s opts="{{ hdd_vg.stdout }} /dev/{{ ssd_drive }}"
  when: cache_vol|failed
```

**Comment by p53** [comment]
will be fixed in pull request #2007, i also prepared module for managing lvol caches  #2041, you can check it/try out

**Comment by hryamzik** [comment]
Thanks! I'll keep an eye on the PR. Not sure about testing as my roles are already done with a workaround. =)

P.S.: you can find cache role [here](https://github.com/gitinsky/ansible-role-dm-cache/blob/master/tasks/main.yml)

**Comment by nitzmahone** [comment]
Apparently fixed in devel by #2007
